requirements.txt

```diff
@@ -4,3 +4,4 @@ matplotlib
 torch>=2.3.0
 pytest>=8.3.5
 PyYAML>=6.0.2 # Required by iree.compiler.dialects.linalg
+pre-commit # For pre-commit hooks (code formatting etc)
```

Comment:
Might want to separate dev requirements from install or CI requirements in the future. Fine for now given the purpose of this repo.